### PR TITLE
java, improvement: error types

### PR DESCRIPTION
**Pull request by dcb6**
We now generate Exception types for all errors that are defined in the IR. Generated clients with an error discrimination strategy of "status code" will throw one of these typed Exceptions based on the status code of error responses. 

Example error type:
  ```java
  public final class MyCompanyAuthBadRequest extends MyCompanyAuthApiError {
    public MyCompanyAuthBadRequest(Object body) {
        super("BadRequest", 400, body);
    }
  }
  ```
Suggest looking at `CustomAuthClient.java` to see the changes and their variations.

### CODE REVIEW

File: seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+    public SeedCustomAuthUnauthorizedRequest(UnauthorizedRequestErrorBody body) {
+        super("UnauthorizedRequest", 401, body);
+        this.body = body;
+    }
+
+    @Override
+    public UnauthorizedRequestErrorBody body() {
+        return this.body;
+    }
```
Comment (dsinghvi) [inline seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java:20]: @dcb6 what happens when you `toString` one of these errors?

Comment (dsinghvi) [reply seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java:20]: or `System.out.printlin(e)`

Comment (dcb6) [reply seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java:20]: Same thing right? 
```java
    @Test
    public void test() {
        // Add tests here and mark this file in .fernignore
        UnauthorizedRequestErrorBody body = UnauthorizedRequestErrorBody.builder().message("Oh no!").build();
        SeedCustomAuthUnauthorizedRequest error = new SeedCustomAuthUnauthorizedRequest(body);
        System.out.println(error);
        assert true;
    }
```
prints:
```
SeedCustomAuthApiError{statusCode: 401, body: {
  "message" : "Oh no!"
}}
```

Comment (dcb6) [reply seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java:20]: was wondering if we should just do a `writeValueAsString` with an object mapper instead?

**Review by dsinghvi — APPROVED** [review]

File: seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java (hunk 1 above)
Comment (dcb6) [reply seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java:20]: I think the best way to improve the `toString` is just to add a message field like so:
```java
public class SeedCustomAuthApiError extends SeedCustomAuthError {

    private final String message;
    private final int statusCode;

    private final Object body;

    public SeedCustomAuthApiError(String message, int statusCode, Object body) {
        super(message);
        this.message = message;
        this.statusCode = statusCode;
        this.body = body;
    }

    public int statusCode() {
        return this.statusCode;
    }

    public Object body() {
        return this.body;
    }

    @java.lang.Override
    public String toString() {
        return "SeedCustomAuthApiError{" + "statusCode: " + statusCode + ", body: " + body +", message: " + message + "}";
    }
}
```

Not exactly sure how i'd cleanly refactor the `createToStringMethod` here to instead call getMessage() if we wanted to go that route: https://github.com/fern-api/fern/blob/dc03bc01865cfdeb9640b4b88d5dae106f0c36da/generators/java/generator-utils/src/main/java/com/fern/java/ObjectMethodFactory.java#L87

Comment (dcb6) [reply seed/java-sdk/custom-auth/src/main/java/com/seed/customAuth/resources/errors/errors/SeedCustomAuthUnauthorizedRequest.java:20]: `SeedCustomAuthError` as is prints`com.seed.customAuth.core.SeedCustomAuthError: Oh no!` for example (roughtly, `<className> <message>`)

so i don't think any improvements are needed there (i believe you mentioned adding a `message` field)

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**